images/sds-drbd-controller/pkg/controller/drbd_storage_class_watcher.go

```diff
@@ -0,0 +1,349 @@
+	NonOperationalByStoragePool     = "storage.deckhouse.io/non-operational-invalid-storage-pool-selected"
+	NonOperationalByZonesLabel      = "storage.deckhouse.io/non-operational-invalid-zones-selected"
+	NonOperationalByReplicasLabel   = "storage.deckhouse.io/non-operational-not-enough-nodes-in-zones"
+)
+
+func NewDRBDStorageClassWatcher(
+	mgr manager.Manager,
+	lc *lapi.Client,
+	interval int,
+	log logger.Logger,
+) (controller.Controller, error) {
+	cl := mgr.GetClient()
+	ctx := context.Background()
+
+	c, err := controller.New(DrbdStorageClassWatcherCtrlName, mgr, controller.Options{
+		Reconciler: reconcile.Func(func(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+			return reconcile.Result{}, nil
+		}),
+	})
+
+	if err != nil {
+		log.Error(err, fmt.Sprintf("[NewDRBDStorageClassWatcher] unable to create controller, name: %s", DrbdStorageClassWatcherCtrlName))
+	}
+
+	go func() {
+		for {
+			time.Sleep(time.Second * time.Duration(interval))
+			log.Info("[NewDRBDStorageClassWatcher] starts reconciliation loop")
+
+			dscs, err := GetAllDRBDStorageClasses(ctx, cl)
+			if err != nil {
+				log.Error(err, "[NewDRBDStorageClassWatcher] unable to get all DRBDStorageClasses")
+				continue
+			}
+
+			sps, err := GetAllLinstorStoragePools(ctx, lc)
+			if err != nil {
+				log.Error(err, "[NewDRBDStorageClassWatcher] unable to get all Linstor Storage Pools")
+			}
+
+			nodeList, err := GetAllKubernetesNodes(ctx, cl)
+			if err != nil {
+				log.Error(err, "[NewDRBDStorageClassWatcher] unable to get all Kubernetes nodes")
+			}
+
+			storagePoolsNodes := SortNodesByStoragePool(nodeList, sps)
+			dspZones := GetDRBDStoragePoolsZones(storagePoolsNodes)
+
+			healthyDSCs := ReconcileDRBDStorageClassPools(ctx, cl, log, dscs, sps)
+			healthyDSCs = ReconcileDRBDStorageClassZones(ctx, cl, log, healthyDSCs, dspZones)
+			ReconcileDRBDStorageClassReplication(ctx, cl, log, healthyDSCs, storagePoolsNodes)
+
+			log.Info("[NewDRBDStorageClassWatcher] ends reconciliation loop")
+		}
+	}()
+
+	return c, err
+}
+
+func SortNodesByStoragePool(nodeList *v1.NodeList, sps map[string][]lapi.StoragePool) map[string][]v1.Node {
+	nodes := make(map[string]v1.Node, len(nodeList.Items))
+	for _, node := range nodeList.Items {
+		nodes[node.Name] = node
+	}
+
+	result := make(map[string][]v1.Node, len(nodes))
+
+	for _, spd := range sps {
+		for _, sp := range spd {
+			result[sp.StoragePoolName] = append(result[sp.StoragePoolName], nodes[sp.NodeName])
+		}
+	}
+
+	return result
+}
+
+func ReconcileDRBDStorageClassPools(
+	ctx context.Context,
+	cl client.Client,
+	log logger.Logger,
+	dscs map[string]v1alpha1.DRBDStorageClass,
+	sps map[string][]lapi.StoragePool,
+) map[string]v1alpha1.DRBDStorageClass {
+	healthy := make(map[string]v1alpha1.DRBDStorageClass, len(dscs))
+	for _, dsc := range dscs {
+		if _, exist := sps[dsc.Spec.StoragePool]; exist {
+			healthy[dsc.Name] = dsc
```

Comment:
I think we need this one
```suggestion
			healthy[dsc.Name] = dsc
			removeNonOperationalLabelOnStorageClass(ctx, cl, log, dsc, NonOperationalByStoragePool)
```

Reply:
Yes, true. Thanks.